nearswap/src/storage_management.rs

```diff
@@ -0,0 +1,80 @@
+use near_contract_standards::storage_management::{
+    StorageBalance, StorageBalanceBounds, StorageManagement,
+};
+use crate::*;
+
+/// Implements users storage management for NearSwap.
+#[near_bindgen]
+impl StorageManagement for NearSwap {
+
+    // Register the caller and store minimal deposit.
+    #[payable]
+    fn storage_deposit(
+        &mut self,
+        registration_only: Option<bool>,
+    ) -> StorageBalance {
+        let amount = env::attached_deposit();
+        let account_id = env::predecessor_account_id();
+        let registration_only = registration_only.unwrap_or(false);
+        let min_balance = self.storage_balance_bounds().min.0;
+        if amount < min_balance {
+            env::panic(ERR12_NOT_ENOUGH_NEAR);
+        }
+        if registration_only {
+            // Registration only setups the account but doesn't leave space for tokens.
+            if self.deposits.contains_key(&account_id) {
+                log!(ERR22_ACC_ALREADY_REGISTERED);
+                if amount > 0 {
+                    Promise::new(env::predecessor_account_id()).transfer(amount);
+                }
+            } else {
+                self.deposit_near(registration_only);
+                let refund = amount - min_balance;
+                if refund > 0 {
+                    Promise::new(env::predecessor_account_id()).transfer(refund);
+                }
```

Comment:
here we should register `min_balance`. The `deposit_near` function doesn't do that. Maybe:
* here, let's don't use `deposit_near`, and instead create and save a new `AccountDeposit` 
* rollback changes from `deposit_near` (below you can call `self.deposit_near()`)
* `deposit_near` can return `AccountDeposit` object - this way we don't need to call `self.storage_balance_of` again in the last line.

Reply:
we don't need to register min_balance here because if user is registering for first time,
```
if !registration_only {
            d.near += amount;
}
```
deposit_near function won't update the amount i.e it will remain `0`.
I have added the check for same in deposit_near function.

Reply:
We shouldn't change the public API of `deposit_near` - adding an option there doesn't make sense. Users shouldn't use that function for registration.

Reply:
ok

Reply:
Updated: 
- Rollbacked changes for deposit_near function
- create new account deposit and save it in lookup map